packages/shared/src/components/modals/streaks/StreakRecoverPopup.spec.tsx

```diff
@@ -0,0 +1,323 @@
+
+    // and rendered
+    const popup = screen.queryByTestId('streak-recover-modal-heading');
+    expect(popup).toBeInTheDocument();
+  });
+});
+
+it('Should have no cost for first time recovery', async () => {
+  mockRecoveryQuery({
+    canDo: true,
+    cost: 0,
+    oldStreakLength: 10,
+  });
+
+  renderComponent({
+    user: {
+      ...loggedUser,
+      reputation: 10,
+    },
+  });
+
+  await waitForNock();
+
+  // rendered
+  const popupHeader = screen.queryByTestId('streak-recover-modal-heading');
+  expect(popupHeader).toBeInTheDocument();
+
+  // expect cost to be 0
+  const cost = screen.getByText('0 Rep');
+  expect(cost).toBeInTheDocument();
+});
+
+it('Should have cost of 25 points for 2nd+ time recovery', async () => {
+  mockRecoveryQuery({
+    canDo: true,
+    cost: 25,
+    oldStreakLength: 10,
+  });
+
+  renderComponent({
+    user: {
+      ...loggedUser,
+      reputation: 50,
+    },
+  });
+
+  await waitForNock();
+
+  // rendered
+  const popupHeader = screen.queryByTestId('streak-recover-modal-heading');
+  expect(popupHeader).toBeInTheDocument();
+
+  // expect cost to be 25
+  const cost = screen.getByText('25 Rep');
+  expect(cost).toBeInTheDocument();
+});
+
+it('Should show not enough points message if user does not have enough points', async () => {
+  mockRecoveryQuery({
+    canDo: true,
+    cost: 25,
+    oldStreakLength: 10,
+  });
+
+  renderComponent({
+    user: {
+      ...loggedUser,
+      reputation: 10,
+    },
+  });
+
+  await waitForNock();
+
+  // rendered
+  const popupHeader = screen.queryByTestId('streak-recover-modal-heading');
+  expect(popupHeader).toBeInTheDocument();
+
+  // expect not enough points message
+  const button = screen.queryByTestId('streak-recover-button');
+  expect(button).not.toBeInTheDocument();
+
+  const copy = screen.queryByTestId('streak-recovery-copy');
+  expect(copy).toHaveTextContent('You don’t have enough');
+});
+
+it('Should show success message on recover', async () => {
+  window.scrollTo = jest.fn();
+
+  mockRecoveryQuery({
+    canDo: true,
+    cost: 25,
+    oldStreakLength: 102,
+  });
+
+  renderComponent({
+    user: {
+      ...loggedUser,
+      reputation: 50,
+    },
+  });
+
+  await waitForNock();
+
+  // rendered
+  const popupHeader = screen.queryByTestId('streak-recover-modal-heading');
+  expect(popupHeader).toBeInTheDocument();
+
+  // button is there
+  const button = screen.getByTestId('streak-recover-button');
+  expect(button).toBeInTheDocument();
+  fireEvent.click(button);
+
+  let mutationCalled = false;
+  mockRecoveryMutation({}, () => {
+    mutationCalled = true;
+  });
+
+  await waitForNock();
+
+  // expect mutation to be called
+  expect(mutationCalled).toBeTruthy();
+
+  // expect success message
+  const alert = await screen.findByRole('alert');
+  expect(alert).toContainHTML('Lucky you! Your streak has been restored');
+});
+
+// it('Should show error message on recover fail', async () => {
```

Comment:
this one can be removed?

Reply:
I commented because I wanted to ask product team if add the error alert, but forgot to ask in the end 😅